src/main/scala/part3typesdatasets/Datasets.scala

```diff
@@ -35,7 +35,7 @@ object Datasets extends App {
                 Horsepower: Option[Long],
                 Weight_in_lbs: Long,
                 Acceleration: Double,
-                Year: Date,
+                Year: String,
                 Origin: String
                 )
 
```

Comment:
Changed Year parameter from Date to String, because an exception was coming.
Exception in thread "main" org.apache.spark.sql.AnalysisException: [CANNOT_UP_CAST_DATATYPE] Cannot up cast Year from "STRING" to "DATE". now it's working fine.

Reply:
Ran and tested all the services, and everything is working well now.
However, while running each and every scala file, below are just the environment-related exceptions which is fine.

RDDs.scala, DataSources.scala - were having exceptions like:
Exception in thread "main" java.lang.RuntimeException: java.io.FileNotFoundException: Hadoop bin directory does not exist: C:\BigDataLocalSetup\hadoop\bin\bin -see https://wiki.apache.org/hadoop/WindowsProblems

Joins.scala - Exception in thread "main" org.postgresql.util.PSQLException: FATAL: password authentication failed for user "docker".

TaxiApplication.scala - Exception in thread "main" org.apache.spark.sql.AnalysisException: [PATH_NOT_FOUND] Path does not exist:file:/C:/Users/brain/rockTheJVM/spark-essentials/path/to/your/dataset/NYC_taxi_2009-2016.parquet.

Everything is working file, please check from your end once.

Reply:
Let's include a section in the README.md file with quick instructions on how to fix it the environment issues on Windows.